homeassistant/components/backup/util.py

```diff
@@ -451,6 +461,8 @@ def on_done(error: Exception | None) -> None:
 
     async def wait(self) -> None:
         """Wait for the worker threads to finish."""
+        for worker in self._workers:
+            worker.thread.raise_exc(AbortCipher)
```

Comment:
How do we know that the new worker that should be allowed to complete can complete before it gets aborted?

Reply:
We only call wait after awaiting agents' `async_upload_backup` methods, at that point it's safe to interrupt any lingering worker.
We could mention it in the docstring, or add a separate method for interrupting workers so it's clearer `wait` is destructive.

Reply:
Ok.